sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/db/derived/TransactionTable.kt

```diff
@@ -0,0 +1,106 @@
+package cash.z.ecc.android.sdk.internal.db.derived
+
+import android.database.sqlite.SQLiteDatabase
+import cash.z.ecc.android.sdk.internal.db.queryAndMap
+import cash.z.ecc.android.sdk.model.BlockHeight
+import cash.z.ecc.android.sdk.model.EncodedTransaction
+import cash.z.ecc.android.sdk.model.FirstClassByteArray
+import cash.z.ecc.android.sdk.model.ZcashNetwork
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.withContext
+import java.util.Locale
+
+internal class TransactionTable(private val zcashNetwork: ZcashNetwork, private val sqliteDatabase: SQLiteDatabase) {
+
+    companion object {
+        private val SELECTION_BLOCK_IS_NULL = String.format(
```

Comment:
Formatting: Some `companion object` definitions have a blank line interleaved with the first statements and others don't. Which of the two is the convention?